src/MergePlugin.php

```diff
@@ -230,7 +230,12 @@ public function onDependencySolve(InstallerEvent $event)
             );
             $request->install($link->getTarget(), $link->getConstraint());
         }
-        if ($this->state->isDevMode()) {
+
+        // Issue #113: Check devMode of event rather than our global state.
+        // Composer fires the PRE_DEPENDENCIES_SOLVING event twice for
+        // `--no-dev` operations to decide which packages are dev only
+        // requirements.
+        if ($event->isDevMode()) {
```

Comment:
:-1: Needs to check PluginState::$mergeDev as well. That flag isn't exposed with a getter yet as it is currently part of the `isDevMode()` check.